### DISCUSSION TITLE
(gRPC) How to import .proto files from project dependencies?

### DISCUSSION THREAD

**Discussion by Ckblck** [quarkusio/quarkus#36481] [Q&A]
**Context:**
I'm working on a multi-module Gradle project that uses gRPC, following the guide provided at [https://quarkus.io/guides/grpc-getting-started](https://quarkus.io/guides/grpc-getting-started). The project structure consists of two modules: `module-commons` and `module-app`, each with its own `proto` directory containing `commons.proto` and `app.proto` respectively. Here's the structure:
```
- module-commons
  |- src/main/proto
     |- commons.proto
  |- build.gradle
- module-app
  |- src/main/proto
     |- app.proto
  |- build.gradle
- build.gradle
```

This is the error message I'm encountering when running the `quarkusBuild` task on the `module-app` project:
```
commons.proto: File not found.
app.proto:9:1: Import "commons.proto" was not found or had errors.
app.proto:13:21: "commons.Message" is not defined.
```

I'm encountering difficulties importing `commons.proto` into the `app.proto` file. I've attempted using the `quarkus.generate-code.grpc.scan-for-imports=all` setting, but it doesn't seem to address this issue. Additionally, I couldn't find a way to specify project dependencies for the `quarkus.generate-code.grpc.scan-for-proto` setting.

When I run `gradle dependencies`, I see the project listed as `project ':module-commons'`. Using the typical `<groupId>:<artifactId>` format for local project dependencies like `com.example:module-commons` doesn't work in this context. 

For external dependencies, such as `proto-google-common-protos`, the setting works as expected. However, the challenge lies in importing multi-module proto files.

The only workaround I can think of is to place all proto files in a single directory, but this would result in an unorganized structure, which is not ideal. How can I properly import `commons.proto` from `app.proto` within this multi-module setup?

> [!NOTE]
> The Intellij plugin [Protobuf ](https://plugins.jetbrains.com/plugin/16422-protobuf) successfully imports and lints the proto files. Problem seems to reside during the Quarkus building process.

[a comment by an automated account is not shown]

**Comment by alesj** [comment]
Did you try current `main upstream` code?
Since @cescoffier recently fixed something that sounds like your problem.
